test/extensions/watchdog/profile_action/profile_action_test.cc

```diff
@@ -0,0 +1,380 @@
+#include <memory>
+
+#include "envoy/common/time.h"
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/extensions/watchdog/profile_action/v3alpha/profile_action.pb.h"
+#include "envoy/filesystem/filesystem.h"
+#include "envoy/server/guarddog_config.h"
+#include "envoy/thread/thread.h"
+
+#include "common/common/assert.h"
+#include "common/filesystem/directory.h"
+#include "common/profiler/profiler.h"
+
+#include "extensions/watchdog/profile_action/config.h"
+#include "extensions/watchdog/profile_action/profile_action.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/test_time.h"
+#include "test/test_common/utility.h"
+
+#include "absl/strings/substitute.h"
+#include "absl/synchronization/mutex.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Watchdog {
+namespace ProfileAction {
+namespace {
+
+// We want to make sure the action is tested with both simulated time and real
+// time, to ensure that it works in production, and that it works in the context
+// of integration tests which are much easier to control with simulated time.
+enum class TimeSystemType { Real, Simulated };
```

Comment:
Can you provide more details on why this is necessary? I think generally we want tests to use `SimulatedTimeSystem` to make them deterministic. It seems like this extension creates timers and uses the a TimeSystem to get timestamps for filenames, so it seems to me that using the simulated time should be sufficient for testing.

Furthermore, the docs in `test/README.md` and `test/test_common/test_time.cc`, state that `GlobalTimeSystem` (as used in `makeTimeSystem` when `TimeSystemType == Real`) is slated to switch to always creating a `SimulatedTimeSystem`. At that point, this parameterization will be no longer change the test's behavior.

Reply:
Great point, and good to know where moving towards just testing with just `SimulatedTimeSystem.` 

The only benefit likely that we'd get from `RealTimeSystem` would be to have the profiler to run for a real duration (vs start then more or less stop it) but that shouldn't be the focus of these tests. 

Given that, I've removed `RealTimeSystem` and parameterization.